components/textfield/index.css

```diff
@@ -143,64 +148,106 @@ governing permissions and limitations under the License.
-    }
+    /* removes the side padding to align the text properly */
+    padding-inline-start: var(--spectrum-textfield-quiet-padding-x);
+    padding-inline-end: var(--spectrum-textfield-quiet-padding-x);
+
+    /* Treat all quiet inputs and textareas the same */
+    resize: none;
+    overflow-y: hidden;
   }
 }
 
-.spectrum-Textfield--multiline {
-  block-size: var(--spectrum-textarea-height);
-  min-block-size: var(--spectrum-textarea-min-height);
-  padding-block: var(--spectrum-textarea-padding-top-adjusted) var(--spectrum-textarea-padding-bottom-adjusted);
-  padding-inline: var(--spectrum-textarea-padding-x-adjusted) var(--spectrum-textarea-padding-x-adjusted);
+.spectrum-Textfield-validationIcon {
+  /* TODO: Confirm if this is ok for the validation icon sizing
+    Note that the sizes are a bit different when compared with old background icons(more noticable for checkmark)
+  */
+  position: absolute;
+  padding-inline-end: calc(var(--spectrum-textfield-padding-x, var(--spectrum-global-dimension-size-150)) / 2);
+  pointer-events: all;
 
-  /* Remove the default vertical scrollbar for textarea in IE. */
-  overflow: auto;
-}
+  .spectrum-Textfield--quiet & {
+    padding-inline-end: 0;
+  }
 
-.spectrum-Textfield--quiet {
-  border-radius: var(--spectrum-textfield-quiet-border-radius);
-  border-width: logical 0 0 var(--spectrum-textfield-quiet-border-size) 0;
+  .spectrum-Textfield.is-invalid & {
+    inline-size: var(--spectrum-icon-alert-medium-width);
+    block-size: var(--spectrum-icon-alert-medium-height);
+    inset-block-end: calc(calc(var(--spectrum-textfield-icon-frame) / 2) - calc(var(--spectrum-icon-alert-medium-height) / 2));
+    inset-inline-end: calc(calc(var(--spectrum-textfield-icon-frame) / 2) - calc(var(--spectrum-icon-alert-medium-width) / 2));
+  }
 
-  /* removes the side padding to align the text properly */
-  padding-inline-start: var(--spectrum-textfield-quiet-padding-x);
-  padding-inline-end: var(--spectrum-textfield-quiet-padding-x);
+  .spectrum-Textfield.is-valid & {
+    inline-size: var(--spectrum-icon-checkmark-medium-width);
+    block-size: var(--spectrum-icon-checkmark-medium-height);
+    inset-block-end: calc(calc(var(--spectrum-textfield-icon-frame) / 2) - calc(var(--spectrum-icon-checkmark-medium-height) / 2));
+    inset-inline-end: var(--spectrum-global-dimension-size-150);
+  }
+}
 
-  /* Treat all quiet inputs and textareas the same */
-  resize: none;
-  overflow-y: hidden;
+/* styles the left icon for textfield, assumes usage of workflow icon sizing (18px by 18px) */
+.spectrum-Textfield-icon {
+  display: block;
+  position: absolute;
+  block-size: var(--spectrum-icon-info-medium-height);
+  inline-size: var(--spectrum-icon-info-medium-width);
+  inset-inline-start: var(--spectrum-global-dimension-size-100);
+  inset-block-start: var(--spectrum-global-dimension-size-85);
+
+  /* styles the textfield properly if the left icon is provided */
+  & + .spectrum-Textfield-input {
```

Comment:
@LFDanLu Note that we're using the presence of an icon to style the input accordingly using the adjacent sibling selector. This is different than the approach in React where an outer class was added.

Reply:
Took a bit of testing locally, but eventually realized that I had to move the icon before the textfield element for this style to work. I'll note this down as a change to be made in RSP, but otherwise seems good to me.